pkg/plugin/system/system.go

```diff
@@ -45,6 +46,11 @@ type pluginSystem struct {
 	clients       []ClientProvider
 	plugins       *PluginList
 	promptRunner  ioutils.PromptRunner
+	callbackMap   map[string]CustomCommandCallbackFn
```

Comment:
Are we using the key anywhere?

Reply:
Actually I don't see this used anywhere where it's actually required...?

Reply:
Its being used below in `AddCustomCommands`

Reply:
That's unnecessary though, I think you can just delete this...

Reply:
If I remove this from the struct it compiles fine but throws a runtime error of:
```
panic: assignment to entry in nil map

goroutine 1 [running]:
aspect.build/cli/pkg/plugin/system.(*pluginSystem).AddCustomCommands(0x14000132070, 0x1400016e000)
	pkg/plugin/system/system.go:140 +0x210
aspect.build/cli/cmd/aspect/root.NewRootCmd({{0x10089cbe0, 0x14000010010}, {0x10089cc20, 0x14000010018}, {0x10089cc20, 0x14000010020}}, {0x1008b3950, 0x14000132070}, 0x1)
	cmd/aspect/root/root.go:134 +0xdf8
aspect.build/cli/cmd/aspect/root.NewDefaultRootCmd({0x1008b3950, 0x14000132070})
	cmd/aspect/root/root.go:41 +0xe0
main.main()
	cmd/aspect/main.go:47 +0x2ac
```

Reply:
Why?

Good chance I'm miss-reading it, but it seems like the only place we assign a value into this map is also the only place we read from it? In `AddCustomCommands`...

Reply:
It is used twice inside of `AddCustomCommands`. We add to the map when `AddCustomCommands` is called but we access it as the `cobra` `RunE` execution happens. So its happening at a different point in time than the original call. We no longer have access to the `node` that we pulled `customCommandCallback` off of when the `RunE` function is called. So we can loop back through the list of nodes and find the node we need to get the `customCommandCallback` or we can save those callbacks somewhere so we can easily access them when we need them

Reply:
Can't you just use a local variable scoped to the `for` loop though? Something such as the `next` var in pkg/interceptors/run.go?

Reply:
Yeah that worked. Switched it out for that